cvat/apps/quality_control/quality_reports.py

```diff
@@ -2480,8 +2495,347 @@ def _save_reports(self, *, task_report: Dict, job_reports: List[Dict]) -> models
         return db_task_report
 
     def _get_task_quality_params(self, task: Task) -> Optional[ComparisonParameters]:
-        quality_params, _ = models.QualitySettings.objects.get_or_create(task=task)
-        return ComparisonParameters.from_dict(quality_params.to_dict())
+        if task.project:
+            try:
+                quality_params = task.project.quality_settings
+            except models.QualitySettings.DoesNotExist:
+                quality_params = None
+        else:
+            quality_params, _ = models.QualitySettings.objects.get_or_create(task=task)
```

Comment:
The logic here seems inconsistent.

* If the task is within a project, then we use the settings from the project. If there are no such settings, we return `None` (which prevents report generation).
* If the task is not within a project, then we use the settings from the task. If there are no such settings, we create default ones.

Why not use default settings in the first case too?

Reply:
> Why not use default settings in the first case too?

Technically, we could, but it's about the server load and existing projects. Basically, I'd prefer to remove creation of default settings for tasks too, but in the case of GT jobs, there is an explicit action of GT job creation, so there is the opportunity to create settings (during the job creation or the first computation, as it is now). In the case of projects there was no such obvious explicit action, so I decided to make creation of quality settings for project to be so. Other variant - add a migration for existing projects to add settings automatically. I don't like it for several reasons. Just adding default settings as you suggest won't work well too, because it can lead to unexpected results and unnecessary server load for unused functionality in existing projects (as there are many things that can trigger report computation).